provider/resources.go

```diff
@@ -144,6 +145,29 @@ func Provider() tfbridge.ProviderInfo {
 		GitHubOrg:   "hashicorp",
 		Repository:  "https://github.com/pulumi/pulumi-azuread",
 		Config: map[string]*tfbridge.SchemaInfo{
+			"client_certificate_path": {
+				Default: &tfbridge.DefaultInfo{
+					EnvVars: []string{"ARM_CLIENT_CERTIFICATE_PATH"},
+				},
```

Comment:
Are we sure we want to set the defaults via the SDK? Does the upstream provider not handle setting the default from the env itself? This will result in the environment variable value being persisted (encrypted) in state which can break refresh and destroy scenarios.

Reply:
Ah, interesting. There were already some existing configs with env variables, so I followed that pattern. Should those have been defined without the env as well?

Reply:
This is a really common bug when setting up provider configuration as the behaviour is surprising.